src/Razor/src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs

```diff
@@ -32,6 +34,11 @@ public override string ToString()
 
         public bool Equals(AllowedChildTagDescriptor other)
         {
+            if (CaseSensitive)
+            {
+                return AllowedChildTagDescriptorComparer.CaseSensitive.Equals(this, other);
+            }
```

Comment:
I'm really not a fan of having *a special comparer* and we should nuke it from orbit. Specifically, the idea of this is wrong, it's a comparer that compares **EVERYTHING** about the tag helper as case sensitive.

IMO equality should behave in the way its currently defined, and the binder should be the thing that enforces case-sensitivity.

Also, this is really problematic because it's non-reflexive.

Reply:
> Specifically, the idea of this is wrong, it's a comparer that compares EVERYTHING about the tag helper as case sensitive.

That is not the case though. It compares as case sensitive only for properties that makes sense. See https://github.com/aspnet/AspNetCore-Tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.Language/BoundAttributeDescriptorComparer.cs#L54 for example.